inc/drivers/MicroBitFile.h

```diff
@@ -0,0 +1,172 @@
+#ifndef MICROBIT_FILE_H
+#define MICROBIT_FILE_H
+
+#include "mbed.h"
+#include "MicroBitConfig.h"
+#include "MicroBitFileSystem.h"
+#include "ManagedString.h"
+
+#define READ                            MB_READ
+#define WRITE                           MB_WRITE
+#define READ_AND_WRITE                  READ | WRITE
+#define CREATE                          MB_CREAT
+
+class MicroBitFile
+{
+
+    int fileHandle;
+    ManagedString fileName;
+
+    public:
+
+    /**
+      * Creates an instance of a MicroBitFile and creates a new file if required.
+      *
+      * @param fileName the name of the file to create/open.
+      *
+      * @param mode One of: READ, WRITE, READ_AND_WRITE. Defaults to READ_AND_WRITE.
+      */
+    MicroBitFile(ManagedString fileName, int mode = READ_AND_WRITE);
+
+    /**
+      * Seeks to a position in this MicroBitFile instance from the beginning of the file.
+      *
+      * @param offset the number of bytes to seek, relative to the beginning of the file.
+      *
+      * @return the new seek position, MICROBIT_NOT_SUPPORTED if the current file handle is invalid,
+      *         MICROBIT_INVALID_PARAMETER if the given offset is negative.
+      */
+    int setPosition(int position);
+
+    /**
+      * Returns the current position of the seek head for the current file.
+      *
+      * @return the new seek position, MICROBIT_NOT_SUPPORTED if the current file handle is invalid.
+      */
+    int getPosition();
+
+    /**
+      * Writes the given bytes to this MicroBitFile instance at the current position.
+      *
+      * @param bytes a pointer to the bytes to write to this file.
+      *
+      * @param len the number of bytes to write.
+      *
+      * @return The number of bytes written, MICROBIT_NOT_SUPPORTED if the current file
+      *         handle is invalid, MICROBIT_INVALID_PARAMETER if bytes is invalid, or
+      *         len is negative.
+      */
+    int write(const char *bytes, int len);
```

Comment:
Do we need a MicroBitFile::Flush(), given that small writes can be cached?  Or are we trying to discourage it?  It guess it could be achieved by doing a close() immediately followed by an open().  While developing code it's sometimes useful to ensure your data got persisted before doing something that might crash ...